go/libraries/doltcore/sqle/single_table_info_db.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2020 Liquidata, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sqle
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/liquidata-inc/go-mysql-server/sql"
+
+	"github.com/liquidata-inc/dolt/go/libraries/doltcore/doltdb"
+	"github.com/liquidata-inc/dolt/go/libraries/doltcore/schema"
+	sqleSchema "github.com/liquidata-inc/dolt/go/libraries/doltcore/sqle/schema"
+	"github.com/liquidata-inc/dolt/go/store/types"
+)
+
+type SingleTableInfoDatabase struct {
+	tableName   string
+	sch         schema.Schema
+	foreignKeys []doltdb.ForeignKey
+	parentSchs  map[string]schema.Schema
+}
+
+var _ sql.Database = (*SingleTableInfoDatabase)(nil)
+var _ sql.Table = (*SingleTableInfoDatabase)(nil)
+var _ sql.IndexedTable = (*SingleTableInfoDatabase)(nil)
+var _ sql.ForeignKeyTable = (*SingleTableInfoDatabase)(nil)
+
+func NewSingleTableDatabase(tableName string, sch schema.Schema, foreignKeys []doltdb.ForeignKey, parentSchs map[string]schema.Schema) *SingleTableInfoDatabase {
```

Comment:
Should have a comment about the intended use case of this DB implementation. Either here or on the struct